block/retriever.go

```diff
@@ -99,19 +99,54 @@ func (m *Manager) processNextDABatch(ctx context.Context, daHeight uint64) error
 	return nil
 }
 
-func (m *Manager) fetchBatch(daHeight uint64) (da.ResultRetrieveBatch, error) {
+func (m *Manager) fetchBatch(daMetaData *da.DAMetaData) (da.ResultRetrieveBatch, error) {
 	var err error
-	batchRes := m.retriever.RetrieveBatches(daHeight)
+	availRes := m.retriever.CheckBatchAvailability(daMetaData)
+
+	switch availRes.Code {
+	case da.StatusUnableToGetProof:
+		//Is not possible to obtain proofs for the specific commitment in the specific height.
+		//This means there is no matching blob for that commitment
+		err = fmt.Errorf("Unable to get proof on height %d for the commitment", daMetaData.Height)
+
+	case da.StatusProofNotMatching:
+		//The proofs are obtained for the commitment, but not matching with the span (index, length) committed to the Hub.
+		err = fmt.Errorf("Span not matching the commitment")
+	case da.StatusError:
+		//There's been an issue validating proofs
+		err = fmt.Errorf("Error validating batch")
+	}
+	if !availRes.DataAvailable {
+		err = fmt.Errorf("no batches found on height %d", daMetaData.Height)
+		//There is no point on fetching the data
+		batchRes := da.ResultRetrieveBatch{}
+		batchRes.Code = da.StatusError
+		batchRes.Message = "Error validating data"
+		batchRes.MetaData = availRes.MetaData
+		return batchRes, err
+
+	}
+	//batchRes.MetaData includes proofs necessary to open disputes with the Hub
+	batchRes := m.retriever.RetrieveBatches(daMetaData)
+	batchRes.MetaData = availRes.MetaData
+
 	switch batchRes.Code {
 	case da.StatusError:
-		err = fmt.Errorf("failed to retrieve batch from height %d: %s", daHeight, batchRes.Message)
+		err = fmt.Errorf("failed to retrieve batch from height %d: %s", daMetaData.Height, batchRes.Message)
```

Comment:
creation of error based on case is a bit dirty IMO. As I see it:
1. we can either return the original Error
2. we can simply log the message and return original error
3. we can have the `message` field holding the error which can be either a string or an error object which we call Error() on.